storage/file_persister_test.go

```diff
@@ -74,3 +79,174 @@ func TestLocalFilePersister(t *testing.T) {
+					"service":"aws_s3",
+					"files":[{"name":"screenshots/some/path/file.png"}]
+				}`,
+			wantPresignedHeaders: map[string]string{
+				"Authorization": "token asd123",
+				"Run_id":        "123456",
+			},
+			uploadResponse:          http.StatusOK,
+			getPresignedURLResponse: http.StatusOK,
+		},
+		{
+			name: "get_presigned_rate_limited",
+			path: "some/path/file.png",
+			data: "here's some data",
+			wantPresignedURLBody: `{
+					"service":"aws_s3",
+					"files":[{"name":"screenshots/some/path/file.png"}]
+				}`,
+			wantPresignedHeaders: map[string]string{
+				"Authorization": "token asd123",
+				"Run_id":        "123456",
+			},
+			getPresignedURLResponse: http.StatusTooManyRequests,
+			wantError:               "get presigned url: status code out of range: request is being rate limited",
+		},
+		{
+			name: "get_presigned_fails",
+			path: "some/path/file.png",
+			data: "here's some data",
+			wantPresignedURLBody: `{
+					"service":"aws_s3",
+					"files":[{"name":"screenshots/some/path/file.png"}]
+				}`,
+			wantPresignedHeaders: map[string]string{
+				"Authorization": "token asd123",
+				"Run_id":        "123456",
+			},
+			getPresignedURLResponse: http.StatusInternalServerError,
+			wantError:               "get presigned url: status code out of range: status code is 500",
+		},
+		{
+			name: "upload_rate_limited",
+			path: "some/path/file.png",
+			data: "here's some data",
+			wantPresignedURLBody: `{
+					"service":"aws_s3",
+					"files":[{"name":"screenshots/some/path/file.png"}]
+				}`,
+			wantPresignedHeaders: map[string]string{
+				"Authorization": "token asd123",
+				"Run_id":        "123456",
+			},
+			uploadResponse:          http.StatusTooManyRequests,
+			getPresignedURLResponse: http.StatusOK,
+			wantError:               "upload status code out of range: request is being rate limited",
+		},
+		{
+			name: "upload_fails",
+			path: "some/path/file.png",
+			data: "here's some data",
+			wantPresignedURLBody: `{
+					"service":"aws_s3",
+					"files":[{"name":"screenshots/some/path/file.png"}]
+				}`,
+			wantPresignedHeaders: map[string]string{
+				"Authorization": "token asd123",
+				"Run_id":        "123456",
+			},
+			uploadResponse:          http.StatusInternalServerError,
+			getPresignedURLResponse: http.StatusOK,
+			wantError:               "upload status code out of range: status code is 500",
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			// This starts the upload server where the files will be uploaded
+			// with the presigned url.
+			uts := httptest.NewServer(http.HandlerFunc(
+				func(w http.ResponseWriter, r *http.Request) {
+					defer func() {
+						require.NoError(t, r.Body.Close())
+					}()
+
+					bb, err := io.ReadAll(r.Body)
+					require.NoError(t, err)
+
+					// Ensures that the data is uploaded to the server and matches
+					// what was sent.
+					assert.Equal(t, tt.data, string(bb))
```

Comment:
I've never used `assert.HTTPBody` and I can't see how it would be useful here. The `HTTPBody` looks like it could be useful to test existing handler implementations where we can provide it with a `method`, `url` and any `url params` (although no body). I may have missed something though.

Reply:
Sorry, may bad. I confused this code without completely understanding it. Resolving.